### PR TITLE
Add checkpoint capability to new diags

### PR DESCRIPTION
**Pull request by MaxThevenet**
While deleting old diags, a number of functions cannot be removed because they are used for checkpoint-restart. This PR doesn't touch the restart capability, but moves the checkpoint capability to the new diags, so that these functions can safely be deleted.

When
```
<diag_name>.diags_type = Full
```
a new flush format "checkpoint" is available
```
<diag_name>.format = checkpoint
```
that will write a checkpoint file. This could be further abstracted so that the user enters
```
<diag_name>.diags_type = checkpoint
```
instead, but I think this minor change will be more relevant once we start working on the back-transformed diagnostics, because then we will clarify what belongs to which `diag_type`.

### CODE REVIEW

File: Source/Diagnostics/FlushFormats/FlushFormatCheckpoint.cpp [hunk 1]
```diff
@@ -0,0 +1,116 @@
+#include "FlushFormatCheckpoint.H"
+#include "WarpX.H"
+#include "Utils/WarpXProfilerWrapper.H"
+
+//#include <AMReX_MultiFabUtil.H>
+//#include <AMReX_PlotFileUtil.H>
+//#include <AMReX_FillPatchUtil_F.H>
```
Comment (ax3l) [inline Source/Diagnostics/FlushFormats/FlushFormatCheckpoint.cpp:7]:
```suggestion
```

File: Source/Diagnostics/FlushFormats/FlushFormatCheckpoint.H [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#ifndef WARPX_FLUSHFORMATCHECKPOINT_H_
+#define WARPX_FLUSHFORMATCHECKPOINT_H_
+
+#include "FlushFormatPlotfile.H"
+
+class FlushFormatCheckpoint : public FlushFormatPlotfile
```
Comment (ax3l) [inline Source/Diagnostics/FlushFormats/FlushFormatCheckpoint.H:6]: Just saw it removed below :D
```suggestion
class FlushFormatCheckpoint final : public FlushFormatPlotfile
```

**Comment by MaxThevenet** [comment]
Thanks for your review, just included your suggestions :).

File: Source/WarpX.H [hunk 3]
```diff
@@ -235,6 +235,7 @@ public:
     const amrex::MultiFab& getBfield_fp  (int lev, int direction) {return *Bfield_fp[lev][direction];}
     const amrex::MultiFab& getrho_fp (int lev) {return *rho_fp[lev];}
     const amrex::MultiFab& getF_fp (int lev) {return *F_fp[lev];}
+    const bool DoPML () const {return do_pml;};
```
Comment (MaxThevenet) [inline Source/WarpX.H:238]:
```suggestion
    bool DoPML () const {return do_pml;};
```
Oops, fix LGTM warning
> The 'const' modifier has no effect on return types

**Pull request merged**